src/App.tsx

```diff
@@ -1,61 +1,148 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoFromServer } from './types/TodoFromServer';
+import { TodoList } from './components/TodoList';
+import { getTodos } from './utils/getTodos';
+import { addTodo } from './utils/addTodo';
+
+const pattern = /[А-ЯёїіA-Z\d\s]/i;
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState('0');
+  const [todos, setTodos] = useState(getTodos(todosFromServer));
+  const [showErrorTitle, setShowErrorTitle] = useState(false);
+  const [showErrorUser, setShowErrorUser] = useState(false);
+
+  function handleChange(
+    func: React.Dispatch<React.SetStateAction<string>>,
+    event: React.ChangeEvent<HTMLInputElement>
+    | React.ChangeEvent<HTMLSelectElement>,
+  ): void {
+    const { value } = event.target;
+    const lastIndex = value.length - 1;
+
+    if (value.length && pattern.test(value[lastIndex])) {
+      func(value);
+    }
+
+    if (!event.target.value) {
+      func(value);
+    }
+  }
+
+  function handleSubmit(
+    event: React.FormEvent<HTMLFormElement>,
+    currTodos: TodoFromServer[],
+  ) {
+    event.preventDefault();
+
+    if (!title || !title.trim()) {
+      setShowErrorTitle(true);
+    }
+
+    if (selectedUserId === '0') {
+      setShowErrorUser(true);
+    }
+
+    if (!title || selectedUserId === '0' || !title.trim()) {
+      return;
+    }
+
+    let maxCurrTodoId = 1;
+
+    for (let i = 1; i < currTodos.length; i += 1) {
+      maxCurrTodoId = Math.max(currTodos[i].id, currTodos[i - 1].id);
+    }
+
+    setTodos(getTodos(
+      addTodo(
+        Number(selectedUserId),
+        maxCurrTodoId + 1,
+        title,
+        currTodos,
+      ),
+    ));
+
+    setTitle('');
+    setSelectedUserId('0');
+  }
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => handleSubmit(
+          event,
+          todos,
```

Comment:
```suggestion
        onSubmit={handleSubmit}
```
use the current state instead of passing todos again. also you don't need to call another function inside a callback. just call your handleSubmit.